contract-examples/contracts/ERC20NativeMinter.sol

```diff
@@ -56,3 +58,147 @@ contract ERC20NativeMinter is ERC20, AllowList {
     return 18;
   }
 }
+
+// TODO:
+// this contract adds another (unwanted) layer of indirection
+// but it's the easiest way to match the previous HardHat testing functionality.
+// Once we completely migrate to DS-test, we can simplify this set of tests.
+contract Minter {
+  ERC20NativeMinter token;
+
+  constructor(address tokenAddress) {
+    token = ERC20NativeMinter(tokenAddress);
+  }
+
+  function mintdraw(uint amount) external {
+    token.mintdraw(amount);
+  }
+
+  function deposit(uint value) external {
+    token.deposit{value: value}();
```

Comment:
is this line the reason the wrapper is necessary? (could we make that clear)?

Reply:
You'll have to be a little more specific about _what_ you want me to clarify. I agree that using this `Minter` contract really muddies the waters regarding clarity. 

To me, it's clear that calling `Minter.deposit(<some_amount>)` is telling an instance of the `Minter` contract to send `<some_amount>` of its ETH to the token it was instantiated with. What's less clear is why we need the indirection. The answer to that is related to the allow-list and testing to make sure that a new address (in this case, the Minter contract) that isn't on the allow-list can receive `<some_amount>` of token. 

Are you more confused yet? haha 

In large part, the reason that this is so confusing is because we are testing the `ERC20NativeMinter` example-contract functionality instead of simply testing the `NativeMinter` precompile directly. This isn't the only place where there exists this indirection between the example-contract and the precompile. 

If we actually isolate contracts for the purpose of testing and contracts that act as examples, the precompile tests will be much easier to understand. The reason I didn't do that right away is because I wanted one PR to migrate as much logic to DS-Test as possible with the exact same coverage. Then another PR (one per precompile would probably be better), to remove the examples entirely where they aren't necessary. The examples should live somewhere else where the only serve as examples, not e2e tests.

With all that said, I propose we leave this as is, and I can start with this one as far as removing examples goes. I left this in another comment somewhere, but the examples should also be split out before this PR lands.